### PR TITLE
Added section on Extension Option for External Logging Solutions

### PR DESCRIPTION
**Pull request by ahardin-rh**
https://trello.com/c/O0WuAw5q/373-document-provide-an-extension-point-to-link-to-other-external-logging-solutions

https://trello.com/c/UWPXQndn/782-2-provide-an-extension-point-to-link-to-other-external-logging-solutions?menu=filter&filter=label:proposed-3.7,label:committed-3.7

### CODE REVIEW

**Comment by ahardin-rh** [comment]
Preview Build: http://file.rdu.redhat.com/~ahardin/07192017/extension-point-external-logging/install_config/web_console_customization.html#extension-option-for-external-logging-solutions

@jwforres PTAL

File: install_config/web_console_customization.adoc [hunk 1]
```diff
@@ -112,7 +109,29 @@ window.OPENSHIFT_EXTENSION_PROPERTIES = {
...
+          extensionRegistry.add('log-links', _.spread(function(resource, options) {
+            return {
+              type: 'dom',
+              node: '<span><a href="https://extension-point.example.com">' + resource.metadata.name + '</a><span class="action-divider">|</span></span>' <1>
+            };
+          }));
+       });
+hawtioPluginLoader.addModule("mylinkextensions");
+----
+<1> The extension file (`\https://extension-point.example.com`, in this example)
```
Comment (jwforres) [inline install_config/web_console_customization.adoc]: "The extension file" doesn't make sense to me here.  Its the URL to the logging stack you are wanting to access.

**Comment by jwforres** [comment]
I would probably add some detail about what the two arguments are (resource and options)

resource is the object that the logs are for, such as a pod, deployment config, or build

options depends on the kind of object but will contain info like the specific container to pull logs for (options.container) or which replication controller the logs are for when getting logs for a deployment config (options.version)

**Comment by ahardin-rh** [comment]
@jwforres Thank you! This is now updated.

@openshift/team-documentation Please peer review. Note that I initially had numbered call-outs, but removed them for a cleaner look.

**Comment by mburke5678** [comment]
@ahardin-rh One comment.

File: install_config/web_console_customization.adoc [hunk 2]
```diff
@@ -112,7 +109,38 @@ window.OPENSHIFT_EXTENSION_PROPERTIES = {
...
+(`\https://extension-point.example.com`, in this example) must be a valid HTTPS
+URL.
+
+The `resource` argument is the object that the logs are for (for example, a pod,
+deployment configuration, or build).
+
+The `options` argument depends on the kind of object, but will contain
+information like the specific container to pull logs for (`options.container`),
+or which replication controller the logs are for when getting logs for a
+deployment configuration (`options.version`).
```
Comment (bfallonf) [inline install_config/web_console_customization.adoc]: Furthering Michael's comment, which part of the example above are the two points referring to? Line 123? Should they be in variable brackets? ( <resource, options> )

cc @jwforres

Comment (ahardin-rh) [inline install_config/web_console_customization.adoc]: Good questions. @jwforres Can you please provide guidance?

**Comment by jwforres** [comment]
@spadgett

File: install_config/web_console_customization.adoc (hunk 2 above)
Comment (spadgett) [inline install_config/web_console_customization.adoc]: The `options` object are the log options we pass to the log REST API endpoint. You've documented the two important options:

For pods, `options.container` specifies the pod container name (e.g., `"my-container"`)
For deployment configs, `options.version` specified the deployment version number (e.g., `"2"`)

**Comment by ahardin-rh** [comment]
@spadgett Thanks! This is updated.

**Comment by xingxingxia** [comment]
LGTM, except some updates are needed:
a. Like other section "Customizing the Logo", this PR's "Extension Option for External Logging Solutions"
is also one supported use case of "Extension Script".
So it should not be put as sub-section under "Loading Extension Scripts and Stylesheets".
Rather, it should be as same level as "Customizing the Logo".

b. For other use case of Extention Script, there is some guide text "Save this script to a file (for example, ...) and add it to the master configuration file".
So for this PR, should also tell user to save it and add into master configuration file.
Otherwise, reader doesn't directly know how to use the the script.

**Comment by adellape** [comment]
[Document: Update OCP ansible installer to install Ansible Service Broker.[asb]](https://trello.com/c/AQfvTlX7/557-document-update-ocp-ansible-installer-to-install-ansible-service-brokerasb)

**Comment by ahardin-rh** [comment]
@xingxingxia Thank you! This is now updated.

**Comment by xingxingxia** [comment]
This PR LGTM.
BTW, I search "save" items in the doc, some mentions restart master, but most don't. Better to make consistency. All need restart master to have the modified master-config take effect. Anyway, this could be another PR if you'd like

**Comment by ahardin-rh** [comment]
@xingxingxia Good catch. Updated.

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../install_config/web_console_customization.adoc#install-config-web-console-customization[Customizing the Web Console]
|Added a new xref:../install_config/web_console_customization.adoc#extension-option-for-external-logging-solutions[Extension Option for External Logging Solutions] section.

**Pull request merged**